daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/AuthIT.scala

```diff
@@ -1,74 +1,77 @@
 // Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package com.daml.lf.engine.script.test
+package com.daml.lf.engine.script
+package test
 
+import com.daml.ledger.api.domain
 import com.daml.ledger.api.testing.utils.SuiteResourceManagementAroundAll
-import com.daml.lf.data.ImmArray
-import com.daml.lf.data.Ref._
+import com.daml.lf.data.{ImmArray, Ref}
 import com.daml.lf.engine.script.ledgerinteraction.ScriptTimeMode
 import com.daml.lf.value.Value
-import com.daml.lf.speedy.SValue._
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AsyncWordSpec
 
 import java.io.File
 import scala.concurrent.Future
+import scala.util.{Failure, Success}
 
 final class AuthIT
     extends AsyncWordSpec
     with CantonFixture
     with Matchers
     with SuiteResourceManagementAroundAll {
+  import CantonFixture._
   import AbstractScriptTest._
 
-  // TODO: https://github.com/digital-asset/daml/issues/16539
-  //  The test is broken, the commented `transform` called in the for construct
+  lazy val secret = "secret"
+
+  override protected lazy val authSecret = Some(secret)
+  override protected lazy val darFiles: List[File] = List(stableDarPath)
+  override protected lazy val devMode: Boolean = false
+  override protected lazy val nParticipants: Int = 1
+  override protected lazy val timeMode = ScriptTimeMode.WallClock
+  override protected lazy val tlsEnable: Boolean = false
+
   "Daml Script against authorized ledger" can {
```

Comment:
We change the test to make authentification by user instead parties, as users is the proper way to authenticate.